packages/react-native-editor/ios/Podfile.lock

```diff
@@ -491,7 +491,7 @@ SPEC CHECKSUMS:
   React-runtimeexecutor: cad74a1eaa53ee6e7a3620231939d8fe2c6afcf0
   ReactCommon: cfe2b7fd20e0dbd2d1185cd7d8f99633fbc5ff05
   ReactNativeDarkMode: 6d807bc8373b872472c8541fc3341817d979a6fb
-  RNCMaskedView: dfeba59697c44d36abec79c062aeb1ea343d610d
+  RNCMaskedView: 66caacf33c86eaa7d22b43178dd998257d5c2e4d
```

Comment:
Strangely enough, these are the changes I see when running `npm run native ios` within this branch. I'm not sure which are correct. Could ruby or bundler versions be causing divergence? I actually see this type of diff quite often in this file, but have yet to explore why they occur. 

```
❯ ruby -v
ruby 2.6.4p104 (2019-08-28 revision 67798) [x86_64-darwin20]

❯ bundle --version
Bundler version 2.2.14
```

<img width="632" alt="podfile-lock-changes" src="https://user-images.githubusercontent.com/438664/122420073-00f09700-cf51-11eb-85e6-87abb8748fd3.png">

Reply:
Aha, my versions are: 
```
$ruby --version
ruby 2.6.0p0 (2018-12-25 revision 66547) [x86_64-darwin20]

$ bundle --version
Bundler version 2.2.16
```

@ceyhun , can this by any chance be related to having to use https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/bin/generate-podspecs.sh?

Reply:
`react-native-modal` doesn't seem to have any `.podspec` or any native code at all, so running `generate-podspecs.sh` shouldn't be necessary.

Seems like the checksums of `packages/react-native-editor/ios/Pods/Local Podspecs/<package-name>.podspec.json` are different for each of us. 


Can you for example maybe share your `react-native-video.podspec.json` file in that folder, so we can compare?

Mine is: 

<details><summary>react-native-video.podspec.json</summary>
<p>


```json
{
  "name": "react-native-video",
  "version": "5.0.2",
  "summary": "A <Video /> element for react-native",
  "description": "A <Video /> element for react-native",
  "license": "MIT",
  "authors": {
    "name": "Brent Vatne",
    "email": "brentvatne@gmail.com",
    "url": "https://github.com/brentvatne"
  },
  "homepage": "https://github.com/brentvatne/react-native-video",
  "source": {
    "git": "https://github.com/brentvatne/react-native-video.git",
    "tag": "5.0.2"
  },
  "platforms": {
    "ios": "8.0",
    "tvos": "9.0"
  },
  "static_framework": true,
  "dependencies": {
    "React-Core": [

    ]
  },
  "default_subspecs": "Video",
  "subspecs": [
    {
      "name": "Video",
      "source_files": "ios/Video/*.{h,m}"
    },
    {
      "name": "VideoCaching",
      "dependencies": {
        "react-native-video/Video": [

        ],
        "SPTPersistentCache": [
          "~> 1.1.0"
        ],
        "DVAssetLoaderDelegate": [
          "~> 0.3.1"
        ]
      },
      "source_files": "ios/VideoCaching/**/*.{h,m}"
    }
  ]
}
```

</p>
</details>

Reply:
(note: both files pasted below include an empty line at the end)

Here's mine (I'm including the full path name for clarity)
<details>
<summary>gutenberg-mobile-root/gutenberg/packages/react-native-editor/ios/Pods/Local Podspecs/react-native-video.podspec.json</summary>
{
  "name": "react-native-video",
  "version": "5.0.2",
  "summary": "A <Video /> element for react-native",
  "description": "A <Video /> element for react-native",
  "license": "MIT",
  "authors": "Brent Vatne <brentvatne@gmail.com> (https://github.com/brentvatne)",
  "homepage": "https://github.com/brentvatne/react-native-video",
  "source": {
    "git": "https://github.com/brentvatne/react-native-video.git",
    "tag": "5.0.2"
  },
  "platforms": {
    "ios": "8.0",
    "tvos": "9.0"
  },
  "static_framework": true,
  "dependencies": {
    "React-Core": [

    ]
  },
  "default_subspecs": "Video",
  "subspecs": [
    {
      "name": "Video",
      "source_files": "ios/Video/*.{h,m}"
    },
    {
      "name": "VideoCaching",
      "dependencies": {
        "react-native-video/Video": [

        ],
        "SPTPersistentCache": [
          "~> 1.1.0"
        ],
        "DVAssetLoaderDelegate": [
          "~> 0.3.1"
        ]
      },
      "source_files": "ios/VideoCaching/**/*.{h,m}"
    }
  ]
}

</details>

The other place I find such a file is:
<details>
<summary>gutenberg-mobile-root/third-party-podspecs/react-native-video.podspec.json</summary>
{
  "name": "react-native-video",
  "version": "5.0.2",
  "summary": "A <Video /> element for react-native",
  "description": "A <Video /> element for react-native",
  "license": "MIT",
  "authors": "Brent Vatne <brentvatne@gmail.com> (https://github.com/brentvatne)",
  "homepage": "https://github.com/brentvatne/react-native-video",
  "source": {
    "git": "https://github.com/brentvatne/react-native-video.git",
    "tag": "5.0.2"
  },
  "platforms": {
    "ios": "8.0",
    "tvos": "9.0"
  },
  "static_framework": true,
  "dependencies": {
    "React-Core": []
  },
  "default_subspecs": "Video",
  "subspecs": [
    {
      "name": "Video",
      "source_files": "ios/Video/*.{h,m}"
    },
    {
      "name": "VideoCaching",
      "dependencies": {
        "react-native-video/Video": [],
        "SPTPersistentCache": [
          "~> 1.1.0"
        ],
        "DVAssetLoaderDelegate": [
          "~> 0.3.1"
        ]
      },
      "source_files": "ios/VideoCaching/**/*.{h,m}"
    }
  ],
  "__WARNING!__": "This file is autogenerated by generate-podspecs.sh script. Do not modify manually. Re-run the script if necessary."
}

</details>

In the meantime, I've [reverted the Podfile.lock changes](https://github.com/WordPress/gutenberg/pull/32772/commits/764f2e5be3ffd811df61d3e9c510243b85e64156) so we can merge the rest of the fixes.

Reply:
Here is my copy of `packages/react-native-editor/ios/Pods/Local\ Podspecs/react-native-video.podspec.json`.

<details><summary>react-native-video.podspec.json</summary>

```json
{
  "name": "react-native-video",
  "version": "5.0.2",
  "summary": "A <Video /> element for react-native",
  "description": "A <Video /> element for react-native",
  "license": "MIT",
  "authors": {
    "name": "Brent Vatne",
    "email": "brentvatne@gmail.com",
    "url": "https://github.com/brentvatne"
  },
  "homepage": "https://github.com/brentvatne/react-native-video",
  "source": {
    "git": "https://github.com/brentvatne/react-native-video.git",
    "tag": "5.0.2"
  },
  "platforms": {
    "ios": "8.0",
    "tvos": "9.0"
  },
  "static_framework": true,
  "dependencies": {
    "React-Core": [

    ]
  },
  "default_subspecs": "Video",
  "subspecs": [
    {
      "name": "Video",
      "source_files": "ios/Video/*.{h,m}"
    },
    {
      "name": "VideoCaching",
      "dependencies": {
        "react-native-video/Video": [

        ],
        "SPTPersistentCache": [
          "~> 1.1.0"
        ],
        "DVAssetLoaderDelegate": [
          "~> 0.3.1"
        ]
      },
      "source_files": "ios/VideoCaching/**/*.{h,m}"
    }
  ]
}
```

</details>

Reply:
@hypest `gutenberg-mobile-root/third-party-podspecs/react-native-video.podspec.json` is only used by WPiOS, so it should affect the `Podfile.lock` in gutenberg.

> In the meantime, I've reverted the Podfile.lock changes so we can merge the rest of the fixes.

I think that should be fine.

For `gutenberg/packages/react-native-editor/ios/Pods/Local Podspecs/react-native-video.podspec.json` seems like only the `author` field is different. It could be maybe because you had old `.podspec`s from the old CocoaPods version before RN upgrade in the `Pods` folder. 
Can you please try running these in `gutenberg/packages/react-native-editor/ios` folder:
```sh
rm -rf Pods/
rm -rf vendor/
```
then from top-level `gutenberg-mobile-root` again:
``` 
npm run core ios
```

Is there still a diff in Podfile.lock after these? Did `gutenberg/packages/react-native-editor/ios/Pods/Local Podspecs/react-native-video.podspec.json` stay the same?

@dcalhoun Yours is identical to mine, but I see now that you got the same checksum as the original in your diff for `react-native-video.podspec.json` 😄 So maybe we should try with another one (like `DoubleConversion`). But before that can you please try running the same commands above to see if it'll generate a `Podfile.lock` diff.

Reply:
did:
```
$rm -Rf gutenberg/packages/react-native-editor/ios/Pods
$ rm -Rf gutenberg/packages/react-native-editor/ios/vendor
$ npm run core ios
```
but I get the exact diff as in the commit I [committed originally](https://github.com/WordPress/gutenberg/pull/32772/commits/ce1baea24d3f8dc0eb16b90cad584ba33fe3b514) 😬

Reply:
After pulling the latest and within `gutenberg`, running the following resulted in one change in the `Podfile.lock` for me.

```bash
rm -rf packages/react-native-editor/ios/Pods packages/react-native-editor/iOS/vendor
npm run core ios
```

<img width="506" alt="podfile-lock-changes" src="https://user-images.githubusercontent.com/438664/122439888-aa3f8900-cf61-11eb-9b5b-c83eb3a64752.png">

Reply:
@hypest That's strange 🤔 Can you try one last thing maybe before removing the folders running in `gutenberg/packages/react-native-editor/ios`: 
```sh
git checkout origin/trunk Podfile.lock
bundle exec pod cache clean --all
rm -rf Pods/
rm -rf vendor/
bundle install
bundle exec pod repo remove trunk
bundle exec pod install --repo-update
```

@dcalhoun Can you maybe share the `gutenberg/packages/react-native-editor/ios/Pods/Local Podspecs/FBReactNativeSpec.podspec.json` file? And maybe I guess trying the above as well won't hurt :)

Reply:
Thanks @dcalhoun! Looks like the paths in the `script_phases` is different. And I guess it will be different for everyone because they are dynamically generated via [this](https://github.com/facebook/react-native/blob/v0.64.0/React/FBReactNativeSpec/FBReactNativeSpec.podspec#L49) in [here](https://github.com/facebook/react-native/blob/v0.64.0/scripts/react_native_pods.rb#L191-L198). 

This phase makes the iOS build call [this script](https://github.com/facebook/react-native/blob/v0.64.0/scripts/generate-specs.sh) each time during a build. I've already added a hack in gutenberg-mobile to remove this `script_phases` when we generate the static podspecs to be used by WPiOS [here](https://github.com/wordpress-mobile/gutenberg-mobile/blob/11c7f704c0eebcf7a48b9e797af6431144ee5652/bin/generate-podspecs.sh#L122-L140) and hosted those generated files instead on gutenberg-mobile [here](https://github.com/wordpress-mobile/gutenberg-mobile/tree/11c7f704c0eebcf7a48b9e797af6431144ee5652/third-party-podspecs/FBReactNativeSpec). 

I guess we can do a similar thing in Gutenberg demo app's `Podfile` somewhere [here](https://github.com/WordPress/gutenberg/blob/3ba2108f636cd68fc16fc4c6564ebed8a487fd30/packages/react-native-editor/ios/Podfile#L31-L41) and maybe even host them in Gutenberg instead of Gutenberg Mobile for both Gutenberg Demo app and WPiOS 🤔